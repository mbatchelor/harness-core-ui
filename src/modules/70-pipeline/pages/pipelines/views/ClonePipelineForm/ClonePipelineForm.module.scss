/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

:global(.bp3-portal .bp3-dialog) {
  &.cloneForm {
    &,
    & :global(.bp3-dialog-header) {
      background-color: var(--form-bg) !important;
    }
  }
}

.form {
  width: 350px;
  padding-bottom: var(--spacing-medium);
}

.error {
  align-self: start;
  max-height: 300px;
}
