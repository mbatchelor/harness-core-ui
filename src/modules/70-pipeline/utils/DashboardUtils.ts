/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { TooltipFormatterContextObject } from 'highcharts'

interface PointStats {
  deployments?: {
    failure?: number
    success?: number
    total?: number
  }
  time?: number
}

/* istanbul ignore next */
export const renderTooltipContent = ({
  time,
  failureRate,
  count,
  successCount,
  failureCount
}: {
  time: string | number
  failureRate: string | number
  count?: number
  successCount?: number
  failureCount?: number
}) => {
  return `<div style="padding: 16px; color: white; width: 282px; height: 128px;">
        <div style="display: flex; justify-content: space-between; border-bottom: 0.5px solid rgba(243, 243, 250); padding-bottom: 7px; margin-bottom: 15px;">
          <div style="font-weight: normal; font-size: 12px; line-height: 18px; opacity: 0.8;">${time}</div>
          <div>
            <span style="white-space: pre; font-weight: bold; font-size: 12px; line-height: 18px; opacity: 0.8;">Deployments: </span>
            <span style="font-weight: bold; font-size: 12px; line-height: 18px;">${count}</span>
          </div>
        </div>
        <div style="display: flex; justify-content: space-between;">
          <div>
            <p style="font-weight: 500; font-size: 10px; line-height: 14px; letter-spacing: 0.2px; color: #D9DAE6; margin-bottom: 0px;">Failure Rate</p>
            <p style="font-weight: 600; font-size: 28px; line-height: 38px; color: #FBE6E4;">${failureRate}</p>
          </div>
          <div style="margin-right: 8px;">
            <div style="display: flex; align-items: center; margin-bottom: 6px;">
              <div style="height: 6px; width: 12px; background-color: #5FB34E; border-radius: 16px; display: inline-block; margin-right: 8px"></div>
              <span style="white-space: pre; font-weight: bold; font-size: 12px; line-height: 16px; letter-spacing: 0.2px; opacity: 0.8;">Success </span>
              <span style="font-weight: bold; font-size: 12px; line-height: 16px; letter-spacing: 0.2px;">${successCount}</span>
            </div>
            <div style="display: flex; align-items: center;">
              <div style="height: 6px; width: 12px; background-color: #EE5F54; border-radius: 16px; display: inline-block; margin-right: 8px"></div>
              <span style="white-space: pre; font-weight: bold; font-size: 12px; line-height: 18px; opacity: 0.8;">Failed </span>
              <span style="font-weight: bold; font-size: 12px; line-height: 18px;">${failureCount}</span>
            </div>
          </div>
        </div>
      </div>`
}

/* istanbul ignore next */
export const getTooltip = (currPoint: TooltipFormatterContextObject): string => {
  const custom = currPoint?.series?.userOptions?.custom
  const point: PointStats = custom?.[currPoint.key]
  const time =
    point && point?.time
      ? new Date(point?.time).toLocaleDateString('en-US', { day: 'numeric', month: 'long' })
      : currPoint.x
  let failureRate: string | number = 'Infinity'
  if (point?.deployments?.failure && point.deployments?.total) {
    failureRate = ((point.deployments.failure / point.deployments.total) * 100).toFixed(1) + '%'
  }
  if (point?.deployments?.failure === 0) {
    failureRate = '0'
  }
  return renderTooltipContent({
    time,
    failureRate,
    count: point?.deployments?.total,
    successCount: point?.deployments?.success,
    failureCount: point?.deployments?.failure
  })
}
