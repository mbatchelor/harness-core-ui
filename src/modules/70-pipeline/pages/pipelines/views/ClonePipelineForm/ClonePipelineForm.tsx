/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { defaultTo, set, get } from 'lodash-es'
import produce from 'immer'
import { useHistory, useParams } from 'react-router-dom'
import * as Yup from 'yup'
import type { FormikProps } from 'formik'
import { Button, Dialog, Formik, FormikForm, FormInput, Layout, SelectOption, useToaster } from '@harness/uicore'
import { useStrings } from 'framework/strings'
import { NameIdDescriptionTags } from '@common/components'
import type { ClonePipelineProperties, PMSPipelineSummaryResponse } from 'services/pipeline-ng'
import { useClonePipeline } from 'services/pipeline-ng'
import {
  OrganizationResponse,
  ProjectAggregateDTO,
  useGetOrganizationList,
  useGetProjectAggregateDTOList
} from 'services/cd-ng'
import type { PipelineType, ProjectPathProps } from '@common/interfaces/RouteInterfaces'
import RbacButton from '@rbac/components/Button/Button'
import { ResourceType } from '@rbac/interfaces/ResourceType'
import { PermissionIdentifier } from '@rbac/interfaces/PermissionIdentifier'
import { NameSchema, IdentifierSchema } from '@common/utils/Validation'
import useRBACError, { RBACError } from '@rbac/utils/useRBACError/useRBACError'
import routes from '@common/RouteDefinitions'

import css from './ClonePipelineForm.module.scss'

export interface ClonePipelineFormProps {
  isOpen: boolean
  onClose(e?: React.SyntheticEvent): void
  originalPipeline: Pick<PMSPipelineSummaryResponse, 'name' | 'identifier' | 'description' | 'tags'>
}

export interface FormState extends Required<ClonePipelineProperties> {
  name?: string
  identifier?: string
  description?: string
  tags?: Record<string, string>
}

export function ClonePipelineForm(props: ClonePipelineFormProps): React.ReactElement {
  const { isOpen, onClose } = props
  const { getString } = useStrings()

  return (
    <Dialog
      isOpen={isOpen}
      title={getString('pipeline.clone')}
      className={css.cloneForm}
      enforceFocus={false}
      onClose={onClose}
      canOutsideClickClose={false}
      lazy
    >
      <ClonePipelineFormInternal {...props} />
    </Dialog>
  )
}

export function ClonePipelineFormInternal(props: ClonePipelineFormProps): React.ReactElement {
  const { isOpen, onClose, originalPipeline } = props
  const { getString } = useStrings()
  const formikRef = React.useRef<FormikProps<FormState>>()
  const history = useHistory()
  const { accountId, orgIdentifier, projectIdentifier, module } = useParams<PipelineType<ProjectPathProps>>()
  const { showSuccess, showError } = useToaster()
  const { getRBACErrorMessage } = useRBACError()

  const { mutate: clonePipeline } = useClonePipeline({
    queryParams: {
      accountIdentifier: accountId
    }
  })
  const { data: orgData, loading: orgDataLoading } = useGetOrganizationList({
    queryParams: {
      accountIdentifier: accountId,
      pageSize: 200
    },
    lazy: !isOpen
  })
  const {
    data: projectData,
    loading: projectDataLoading,
    refetch: refetchProjectsData
  } = useGetProjectAggregateDTOList({
    queryParams: {
      accountIdentifier: accountId,
      orgIdentifier
    },
    lazy: !isOpen
  })

  const organizations: SelectOption[] = React.useMemo(() => {
    const data: OrganizationResponse[] = get(orgData, 'data.content', [])
    return data.map(org => {
      return {
        label: org.organization.name,
        value: org.organization.identifier
      }
    })
  }, [orgData])

  const projects: SelectOption[] = React.useMemo(() => {
    const data: ProjectAggregateDTO[] = get(projectData, 'data.content', [])
    return data.map(project => {
      return {
        label: project.projectResponse.project.name,
        value: project.projectResponse.project.identifier
      }
    })
  }, [projectData])

  function handleOrgChange(item: SelectOption): void {
    refetchProjectsData({
      queryParams: {
        accountIdentifier: accountId,
        orgIdentifier: item.value as string
      }
    })

    /* istanbul ignore else */
    if (formikRef.current) {
      const newValues = produce(formikRef.current.values, draft => {
        set(draft, 'destinationConfig.orgIdentifier', item.value)
        set(draft, 'destinationConfig.projectIdentifier', '')
      })
      formikRef.current.setValues(newValues)
    }
  }

  async function handleSubmit(formData: FormState): Promise<FormState> {
    try {
      const data: Required<ClonePipelineProperties> = {
        sourceConfig: { ...formData.sourceConfig },
        destinationConfig: {
          ...formData.destinationConfig,
          pipelineIdentifier: formData.identifier,
          pipelineName: formData.name,
          description: formData.description,
          tags: formData.tags
        },
        cloneConfig: { ...formData.cloneConfig }
      }

      await clonePipeline(data)

      showSuccess(
        getString('pipeline.cloneSuccess', {
          name: originalPipeline.name,
          cloneName: formData.name
        })
      )

      history.push(
        routes.toPipelineDetail({
          module,
          pipelineIdentifier: defaultTo(data.destinationConfig.pipelineIdentifier, ''),
          orgIdentifier: defaultTo(data.destinationConfig.orgIdentifier, ''),
          projectIdentifier: defaultTo(data.destinationConfig.projectIdentifier, ''),
          accountId
        })
      )

      onClose()
    } catch (e: unknown) {
      showError(getRBACErrorMessage(e as RBACError))
    }

    return formData
  }

  const loading = orgDataLoading || projectDataLoading
  const initialvalues: FormState = React.useMemo(
    () => ({
      name: `${originalPipeline.name} - Clone`,
      identifier: `${originalPipeline.identifier}_Clone`,
      tags: originalPipeline.tags,
      description: originalPipeline.description,
      sourceConfig: {
        orgIdentifier,
        projectIdentifier,
        pipelineIdentifier: originalPipeline.identifier
      },
      destinationConfig: {
        orgIdentifier,
        projectIdentifier
      },
      cloneConfig: {
        connectors: false,
        inputSets: false,
        templates: false,
        triggers: false
      }
    }),
    [originalPipeline, orgIdentifier, projectIdentifier]
  )

  return (
    <div data-testid="clone-pipeline-form" onClick={e => e.stopPropagation()}>
      <Formik<FormState>
        formName="clone-pipeline"
        initialValues={initialvalues}
        onSubmit={handleSubmit}
        formLoading={loading}
        validationSchema={Yup.object().shape({
          name: NameSchema(),
          identifier: IdentifierSchema(),
          destinationConfig: Yup.object().shape({
            orgIdentifier: Yup.string().trim().required(getString('validation.orgValidation')),
            projectIdentifier: Yup.string().trim().required(getString('common.validation.projectIsRequired'))
          })
        })}
      >
        {formikProps => {
          formikRef.current = formikProps

          return (
            <FormikForm className={css.form}>
              <NameIdDescriptionTags formikProps={formikProps} />
              <FormInput.Select
                selectProps={{ usePortal: true }}
                label={getString('orgLabel')}
                name="destinationConfig.orgIdentifier"
                items={organizations}
                onChange={handleOrgChange}
              />
              <FormInput.Select
                key={get(formikProps.values, 'destinationConfig.orgIdentifier')}
                selectProps={{ usePortal: true }}
                label={getString('projectLabel')}
                name="destinationConfig.projectIdentifier"
                items={projects}
              />
              <Layout.Horizontal padding={{ top: 'large' }} spacing="medium">
                <RbacButton
                  permission={{
                    resource: { resourceType: ResourceType.PIPELINE },
                    permission: PermissionIdentifier.EDIT_PIPELINE
                  }}
                  intent="primary"
                  text={getString('projectCard.clone')}
                  type="submit"
                  data-testid="clone"
                />
                <Button intent="none" text={getString('cancel')} type="reset" onClick={onClose} />
              </Layout.Horizontal>
            </FormikForm>
          )
        }}
      </Formik>
    </div>
  )
}
