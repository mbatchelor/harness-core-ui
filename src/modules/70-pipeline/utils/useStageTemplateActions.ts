/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { defaultTo } from 'lodash-es'
import { parse } from 'yaml'
import produce from 'immer'
import { useCallback } from 'react'
import type { StageElementConfig } from 'services/cd-ng'
import { usePipelineContext } from '@pipeline/components/PipelineStudio/PipelineContext/PipelineContext'
import { createTemplate, getStageType } from '@pipeline/utils/templateUtils'
import type { TemplateSummaryResponse } from 'services/template-ng'

interface TemplateActionsReturnType {
  addOrUpdateTemplate: (selectedTemplate?: TemplateSummaryResponse) => Promise<void>
  removeTemplate: () => Promise<void>
}

export function useStageTemplateActions(): TemplateActionsReturnType {
  const {
    state: {
      selectionState: { selectedStageId = '' },
      templateTypes
    },
    updateStage,
    getStageFromPipeline,
    getTemplate
  } = usePipelineContext()
  const { stage } = getStageFromPipeline(selectedStageId)

  const addOrUpdateTemplate = useCallback(
    async (selectedTemplate?: TemplateSummaryResponse) => {
      try {
        const { template, isCopied } = await getTemplate({
          templateType: 'Stage',
          selectedChildType: getStageType(stage?.stage, templateTypes),
          selectedTemplate
        })
        const node = stage?.stage
        const processNode = isCopied
          ? produce(defaultTo(parse(template?.yaml || '')?.template.spec, {}) as StageElementConfig, draft => {
              draft.name = defaultTo(node?.name, '')
              draft.identifier = defaultTo(node?.identifier, '')
            })
          : createTemplate(node, template)
        await updateStage(processNode)
      } catch (_) {
        // Do nothing.. user cancelled template selection
      }
    },
    [stage?.stage, templateTypes, getTemplate, updateStage]
  )

  const removeTemplate = useCallback(async () => {
    const node = stage?.stage
    const processNode = produce({} as StageElementConfig, draft => {
      draft.name = defaultTo(node?.name, '')
      draft.identifier = defaultTo(node?.identifier, '')
      draft.type = getStageType(node, templateTypes)
    })
    await updateStage(processNode)
  }, [stage?.stage, templateTypes, updateStage])

  return { addOrUpdateTemplate, removeTemplate }
}
