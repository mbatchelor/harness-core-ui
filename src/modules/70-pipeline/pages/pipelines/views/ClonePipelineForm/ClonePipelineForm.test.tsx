/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import {
  render,
  queryByAttribute,
  fireEvent,
  waitFor,
  findByText as findByTextGlobal,
  act
} from '@testing-library/react'
import { useToaster } from '@harness/uicore'
import { TestWrapper, CurrentLocation } from '@common/utils/testUtils'
import routes from '@common/RouteDefinitions'
import type { PipelineType, ProjectPathProps } from '@common/interfaces/RouteInterfaces'
import { useClonePipeline } from 'services/pipeline-ng'
import { useGetProjectAggregateDTOList } from 'services/cd-ng'

import { ClonePipelineForm, ClonePipelineFormProps } from './ClonePipelineForm'

jest.mock('@harness/uicore', () => ({
  ...jest.requireActual('@harness/uicore'),
  useToaster: jest.fn().mockReturnValue({ showSuccess: jest.fn() })
}))

const originalPipeline: ClonePipelineFormProps['originalPipeline'] = {
  name: 'My Pipeline',
  identifier: 'My_Pipeline',
  description: 'My Pipeline Description',
  tags: { MyTag1: '', MyTag2: '' }
}

const TEST_PATH = routes.toCDProject({
  module: ':module',
  orgIdentifier: ':orgIdentifier',
  projectIdentifier: ':projectIdentifier',
  accountId: ':accountId'
})

const PATH_PARAMS: PipelineType<ProjectPathProps> = {
  module: 'cd',
  orgIdentifier: 'TEST_ORG1',
  projectIdentifier: 'TEST_PROJECT1',
  accountId: 'TEST_ACCOUNT1'
}

const FORM_ID = 'clone-pipeline-form'

jest.mock('services/cd-ng', () => ({
  useGetOrganizationList: jest.fn().mockReturnValue({
    data: {
      data: {
        content: [
          { organization: { name: 'Test Org 1', identifier: 'TEST_ORG1' } },
          { organization: { name: 'Test Org 2', identifier: 'TEST_ORG2' } }
        ]
      }
    },
    loading: false
  }),
  useGetProjectAggregateDTOList: jest.fn().mockReturnValue({
    data: {
      data: {
        content: [
          { projectResponse: { project: { name: 'Test Project 1', identifier: 'TEST_PROJECT1' } } },
          { projectResponse: { project: { name: 'Test Project 2', identifier: 'TEST_PROJECT2' } } }
        ]
      }
    },
    loading: false
  })
}))

jest.mock('services/pipeline-ng', () => ({
  useClonePipeline: jest.fn().mockReturnValue({ mutate: jest.fn() })
}))

describe('<ClonePipelineForm /> tests', () => {
  test('snapshot test', async () => {
    const { findByTestId } = render(
      <TestWrapper path={TEST_PATH} pathParams={PATH_PARAMS as any}>
        <ClonePipelineForm originalPipeline={originalPipeline} onClose={jest.fn()} isOpen />
      </TestWrapper>
    )

    const form = await findByTestId(FORM_ID)
    expect(form).toMatchSnapshot()

    const name = queryByAttribute('name', form, 'name') as HTMLInputElement
    const description = queryByAttribute('name', form, 'description') as HTMLInputElement
    const id = queryByAttribute('class', form, 'bp3-editable-text-content') as HTMLElement
    const org = queryByAttribute('name', form, 'destinationConfig.orgIdentifier') as HTMLInputElement
    const proj = queryByAttribute('name', form, 'destinationConfig.projectIdentifier') as HTMLInputElement

    expect(name.value).toBe('My Pipeline - Clone')
    expect(id.textContent).toBe('My_Pipeline_Clone')
    expect(org.value).toBe('Test Org 1')
    expect(proj.value).toBe('Test Project 1')
    expect(description.value).toBe(originalPipeline.description)
  })

  test('submit flow', async () => {
    const clonePipeline = jest.fn()
    ;(useClonePipeline as jest.Mock).mockImplementation().mockReturnValue({
      mutate: clonePipeline
    })
    const { findByTestId } = render(
      <TestWrapper path={TEST_PATH} pathParams={PATH_PARAMS as any}>
        <ClonePipelineForm originalPipeline={originalPipeline} onClose={jest.fn()} isOpen />
        <CurrentLocation />
      </TestWrapper>
    )

    const clone = await findByTestId('clone')

    act(() => {
      fireEvent.click(clone)
    })

    await waitFor(() =>
      expect(clonePipeline).toHaveBeenLastCalledWith({
        cloneConfig: {
          connectors: false,
          inputSets: false,
          templates: false,
          triggers: false
        },
        destinationConfig: {
          orgIdentifier: 'TEST_ORG1',
          pipelineIdentifier: 'My_Pipeline_Clone',
          pipelineName: 'My Pipeline - Clone',
          projectIdentifier: 'TEST_PROJECT1',
          description: 'My Pipeline Description',
          tags: { MyTag1: '', MyTag2: '' }
        },
        sourceConfig: {
          orgIdentifier: 'TEST_ORG1',
          pipelineIdentifier: 'My_Pipeline',
          projectIdentifier: 'TEST_PROJECT1'
        }
      })
    )
    const newLocation = await findByTestId('location')

    expect(newLocation).toMatchInlineSnapshot(`
      <div
        data-testid="location"
      >
        /account/TEST_ACCOUNT1/cd/orgs/TEST_ORG1/projects/TEST_PROJECT1/pipelines/My_Pipeline_Clone
      </div>
    `)
  })

  test('handles errors', async () => {
    const showError = jest.fn()
    const showSuccess = jest.fn()
    ;(useToaster as jest.Mock).mockReturnValue({ showError, showSuccess })

    const err_msg =
      'Pipeline [test1_Clone_Clone] under Project[defaultproject], Organization [default] already exists or has been deleted.'
    ;(useClonePipeline as jest.Mock).mockImplementation().mockReturnValue({
      mutate: jest.fn().mockRejectedValue({
        data: {
          status: 'ERROR',
          code: 'DUPLICATE_FIELD',
          message: err_msg
        }
      })
    })
    const { findByTestId } = render(
      <TestWrapper path={TEST_PATH} pathParams={PATH_PARAMS as any}>
        <ClonePipelineForm originalPipeline={originalPipeline} onClose={jest.fn()} isOpen />
      </TestWrapper>
    )

    const clone = await findByTestId('clone')

    act(() => {
      fireEvent.click(clone)
    })

    await waitFor(() => expect(showError).toHaveBeenLastCalledWith(err_msg))
  })

  test('new projects are fetched, when org is changed', async () => {
    const refetch = jest.fn()
    ;(useGetProjectAggregateDTOList as jest.Mock).mockImplementation().mockReturnValue({
      refetch,
      data: {
        data: {
          content: [
            { projectResponse: { project: { name: 'Test Project 1', identifier: 'TEST_PROJECT1' } } },
            { projectResponse: { project: { name: 'Test Project 2', identifier: 'TEST_PROJECT2' } } }
          ]
        }
      },
      loading: false
    })

    const { findByTestId } = render(
      <TestWrapper path={TEST_PATH} pathParams={PATH_PARAMS as any}>
        <ClonePipelineForm originalPipeline={originalPipeline} onClose={jest.fn()} isOpen />
      </TestWrapper>
    )

    const form = await findByTestId(FORM_ID)
    const org = queryByAttribute('name', form, 'destinationConfig.orgIdentifier') as HTMLInputElement

    act(() => {
      fireEvent.click(org.parentElement!.querySelector('.bp3-icon')!)
    })

    await waitFor(() => {
      expect(document.querySelector('ul.bp3-menu')).toBeTruthy()
    })

    const item = await findByTextGlobal(document.body, 'Test Org 2')

    act(() => {
      fireEvent.click(item)
    })

    await waitFor(() =>
      expect(refetch).toHaveBeenLastCalledWith({
        queryParams: {
          accountIdentifier: 'TEST_ACCOUNT1',
          orgIdentifier: 'TEST_ORG2'
        }
      })
    )

    const proj = queryByAttribute('name', form, 'destinationConfig.projectIdentifier') as HTMLInputElement

    await waitFor(() => {
      expect(proj.value).toBe('')
    })
  })
})
