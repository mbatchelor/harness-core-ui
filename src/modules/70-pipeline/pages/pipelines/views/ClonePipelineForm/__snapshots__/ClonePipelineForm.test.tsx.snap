// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClonePipelineForm /> tests snapshot test 1`] = `
<div
  data-testid="clone-pipeline-form"
>
  <div
    class="OverlaySpinner--overlaySpinner"
  >
    <form
      action="#"
      class="FormikForm--main form"
    >
      <div>
        <div
          class="StyledProps--font StyledProps--main main"
        >
          <div
            class="InputWithIdentifier--txtNameContainer TextInput--main"
          >
            <div
              class="Layout--horizontal Layout--layout-spacing-xsmall StyledProps--main InputWithIdentifier--txtIdContainer"
            >
              <div
                class="Text--withDocsTooltip"
                data-tooltip-id="universalInputNameWithId"
              >
                <p
                  class="StyledProps--font StyledProps--main"
                >
                  Id
                </p>
              </div>
              <p
                class="StyledProps--font StyledProps--main"
              >
                :
              </p>
              <span
                class="bp3-popover-wrapper"
              >
                <span
                  class="bp3-popover-target"
                >
                  <div
                    class="bp3-editable-text InputWithIdentifier--idValue"
                    tabindex="0"
                  >
                    <span
                      class="bp3-editable-text-content"
                      style="height: 1px; line-height: 1px; min-width: 0;"
                    >
                      My_Pipeline_Clone
                    </span>
                  </div>
                </span>
              </span>
              <span
                class="bp3-icon StyledProps--main"
                data-icon="Edit"
                style="vertical-align: middle; cursor: pointer;"
              >
                <svg
                  height="12"
                  viewBox="0 0 16 16"
                  width="12"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M11.207.485a.5.5 0 01.354-.146 3.5 3.5 0 013.5 3.5.5.5 0 01-.146.353l-11 11a.5.5 0 01-.354.147h-3a.5.5 0 01-.5-.5v-3a.5.5 0 01.146-.354l11-11zm.553.862L1.061 12.046v2.293h2.293l10.7-10.7a2.5 2.5 0 00-2.294-2.292z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M11.89 2.646a.5.5 0 010 .708L8.353 6.889a.5.5 0 01-.708-.707l3.536-3.536a.5.5 0 01.707 0z"
                    fill="currentColor"
                    fill-rule="evenodd"
                    opacity="0.3"
                  />
                </svg>
              </span>
            </div>
            <div
              class="bp3-form-group"
              data-id="name-0"
            >
              <label
                class="bp3-label"
                for="name"
              >
                <span
                  class="Tooltip--acenter"
                  data-tooltip-id="clone-pipeline_name"
                >
                  name
                </span>
                 
                <span
                  class="bp3-text-muted"
                />
              </label>
              <div
                class="bp3-form-content"
              >
                <div
                  class="bp3-input-group"
                >
                  <input
                    autocomplete="off"
                    class="bp3-input"
                    name="name"
                    placeholder="common.namePlaceholder"
                    type="text"
                    value="My Pipeline - Clone"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="StyledProps--font StyledProps--main"
            style="margin-bottom: 0px;"
          >
            <label
              class="Label--main bp3-label descriptionLabel"
            >
              optionalField
            </label>
            <div
              class="bp3-form-group"
              data-id="description-1"
            >
              <div
                class="bp3-form-content"
              >
                <textarea
                  autocomplete="off"
                  class="bp3-input bp3-fill"
                  maxlength="1024"
                  name="description"
                  placeholder="common.descriptionPlaceholder"
                >
                  My Pipeline Description
                </textarea>
              </div>
            </div>
          </div>
          <div
            class="StyledProps--font StyledProps--main"
          >
            <label
              class="Label--main bp3-label descriptionLabel"
            >
              optionalField
            </label>
            <div
              class="bp3-form-group"
              data-id="undefined-2"
            >
              <div
                class="bp3-form-content"
              >
                <div
                  class="bp3-input bp3-tag-input"
                >
                  <div
                    class="bp3-tag-input-values"
                  >
                    <span
                      class="bp3-tag"
                      data-tag-index="0"
                    >
                      <span
                        class="bp3-text-overflow-ellipsis bp3-fill"
                      >
                        MyTag1
                      </span>
                      <button
                        class="bp3-tag-remove"
                        type="button"
                      >
                        <span
                          class="bp3-icon bp3-icon-small-cross"
                          icon="small-cross"
                        >
                          <svg
                            data-icon="small-cross"
                            height="16"
                            viewBox="0 0 16 16"
                            width="16"
                          >
                            <desc>
                              small-cross
                            </desc>
                            <path
                              d="M9.41 8l2.29-2.29c.19-.18.3-.43.3-.71a1.003 1.003 0 00-1.71-.71L8 6.59l-2.29-2.3a1.003 1.003 0 00-1.42 1.42L6.59 8 4.3 10.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71L8 9.41l2.29 2.29c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71L9.41 8z"
                              fill-rule="evenodd"
                            />
                          </svg>
                        </span>
                      </button>
                    </span>
                    <span
                      class="bp3-tag"
                      data-tag-index="1"
                    >
                      <span
                        class="bp3-text-overflow-ellipsis bp3-fill"
                      >
                        MyTag2
                      </span>
                      <button
                        class="bp3-tag-remove"
                        type="button"
                      >
                        <span
                          class="bp3-icon bp3-icon-small-cross"
                          icon="small-cross"
                        >
                          <svg
                            data-icon="small-cross"
                            height="16"
                            viewBox="0 0 16 16"
                            width="16"
                          >
                            <desc>
                              small-cross
                            </desc>
                            <path
                              d="M9.41 8l2.29-2.29c.19-.18.3-.43.3-.71a1.003 1.003 0 00-1.71-.71L8 6.59l-2.29-2.3a1.003 1.003 0 00-1.42 1.42L6.59 8 4.3 10.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71L8 9.41l2.29 2.29c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71L9.41 8z"
                              fill-rule="evenodd"
                            />
                          </svg>
                        </span>
                      </button>
                    </span>
                    <input
                      class="bp3-input-ghost"
                      data-mentions="kv-tag-input-tags"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="bp3-form-group"
          data-id="destinationConfig.orgIdentifier-3"
        >
          <label
            class="bp3-label"
            for="destinationConfig.orgIdentifier"
          >
            <span
              class="Tooltip--acenter"
              data-tooltip-id="clone-pipeline_destinationConfig.orgIdentifier"
            >
              orgLabel
            </span>
             
            <span
              class="bp3-text-muted"
            />
          </label>
          <div
            class="bp3-form-content"
          >
            <div
              class="bp3-popover-wrapper Select--main bp3-fill"
            >
              <div
                class="bp3-popover-target"
              >
                <div
                  class="bp3-input-group"
                >
                  <input
                    autocomplete="off"
                    class="bp3-input"
                    name="destinationConfig.orgIdentifier"
                    placeholder="- Select -"
                    style="padding-right: 0px;"
                    type="text"
                    value="Test Org 1"
                  />
                  <span
                    class="bp3-input-action"
                  >
                    <span
                      class="bp3-icon bp3-icon-chevron-down StyledProps--main StyledProps--padding-small"
                      icon="chevron-down"
                    >
                      <svg
                        data-icon="chevron-down"
                        height="14"
                        viewBox="0 0 16 16"
                        width="14"
                      >
                        <desc>
                          chevron-down
                        </desc>
                        <path
                          d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="bp3-form-group"
          data-id="destinationConfig.projectIdentifier-4"
        >
          <label
            class="bp3-label"
            for="destinationConfig.projectIdentifier"
          >
            <span
              class="Tooltip--acenter"
              data-tooltip-id="clone-pipeline_destinationConfig.projectIdentifier"
            >
              projectLabel
            </span>
             
            <span
              class="bp3-text-muted"
            />
          </label>
          <div
            class="bp3-form-content"
          >
            <div
              class="bp3-popover-wrapper Select--main bp3-fill"
            >
              <div
                class="bp3-popover-target"
              >
                <div
                  class="bp3-input-group"
                >
                  <input
                    autocomplete="off"
                    class="bp3-input"
                    name="destinationConfig.projectIdentifier"
                    placeholder="- Select -"
                    style="padding-right: 0px;"
                    type="text"
                    value="Test Project 1"
                  />
                  <span
                    class="bp3-input-action"
                  >
                    <span
                      class="bp3-icon bp3-icon-chevron-down StyledProps--main StyledProps--padding-small"
                      icon="chevron-down"
                    >
                      <svg
                        data-icon="chevron-down"
                        height="14"
                        viewBox="0 0 16 16"
                        width="14"
                      >
                        <desc>
                          chevron-down
                        </desc>
                        <path
                          d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main StyledProps--padding-top-large"
        >
          <button
            class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--intent-primary Button--with-current-color"
            data-testid="clone"
            type="submit"
          >
            <span
              class="bp3-button-text"
            >
              projectCard.clone
            </span>
          </button>
          <button
            class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color"
            type="reset"
          >
            <span
              class="bp3-button-text"
            >
              cancel
            </span>
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
